src/pytorch_lightning/strategies/fully_sharded_native.py

```diff
@@ -201,6 +206,26 @@ def _configure_launcher(self) -> None:
             self._launcher = _SubprocessScriptLauncher(self.cluster_environment, self.num_processes, self.num_nodes)
             self._rank_0_will_call_children_scripts = True
 
+    def _setup_model(self, model: torch.nn.Module) -> FullyShardedDataParallel:
```

Comment:
This is the signature of this method, so this looks correct.

Reply:
Okay, but then I suggest we use the model passed there instead of relying on `self.lightning_module`

Reply:
Yes I agree. @rohitgr7 You can look how it's done in DDP.

Reply:
it's done like this in DDP too. The reason I didn't change it because it was raising mypy error because of different base class signature.

and something related to Lite is mentioned there:
```
# TODO (@awaelchli): standardize this across all plugins in Lightning and Lite. Related refactor: #7324
```
so I didn't change the signature.

Reply:
> it's done like this in DDP too

I don't get this comment, in `DDPStrategy._setup_model` the input argument is what gets wrapped into `DistributedDataParallel`

Reply:
yes, and here also the input argument is wrapped
```py
return FullyShardedDataParallel(
            module=model,
```